usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateShapeIdReferenceInStopTimeTest.java

```diff
@@ -0,0 +1,293 @@
+        assertEquals("non existing shape id", noticeList.get(0).getNoticeSpecific(KEY_UNKNOWN_SHAPE_ID));
+
+        verifyNoMoreInteractions(mockDataRepo, mockLogger, mockResultRepo, mockStopTime, mockTrip);
+    }
+
+    // suppress warning regarding ignored result of method since it is not necessary here.
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @Test
+    void stopTimeWithNonNullShapeDistTravelledPlusValidTripIdAndNullAndValidShapeIdShouldNotGenerateNotice() {
+        final StopTime mockStopTime = mock(StopTime.class);
+        when(mockStopTime.getTripId()).thenReturn("trip id");
+        when(mockStopTime.getStopSequence()).thenReturn(2);
+        when(mockStopTime.getShapeDistTraveled()).thenReturn(5f);
+        final Map<String, TreeMap<Integer, StopTime>> mockStopTimeCollection = new HashMap<>();
+        final TreeMap<Integer, StopTime> innerMap = new TreeMap<>();
+        innerMap.put(2, mockStopTime);
+        mockStopTimeCollection.put("trip id", innerMap);
+
+        final Trip mockTrip = mock(Trip.class);
+        when(mockTrip.getShapeId()).thenReturn(null);
+        when(mockTrip.getTripId()).thenReturn("trip id");
+        when(mockTrip.getShapeId()).thenReturn("non existing shape id");
+
+        final ShapePoint mockShapePoint = mock(ShapePoint.class);
+        final TreeMap<Integer, ShapePoint> mockShapePointCollection = new TreeMap<>();
+        mockShapePointCollection.put(28, mockShapePoint);
+        mockStopTimeCollection.put("trip id", innerMap);
+
+        final GtfsDataRepository mockDataRepo = mock(GtfsDataRepository.class);
+        when(mockDataRepo.getStopTimeAll()).thenReturn(mockStopTimeCollection);
+        when(mockDataRepo.getStopTimeByTripId("trip id")).thenReturn(innerMap);
+        when(mockDataRepo.getTripById("trip id")).thenReturn(null);
+        when(mockDataRepo.getTripById("trip id")).thenReturn(mockTrip);
+        when(mockDataRepo.getShapeById("shape id")).thenReturn(mockShapePointCollection);
+
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final Logger mockLogger = mock(Logger.class);
+
+        final ValidateShapeIdReferenceInStopTime underTest =
+                new ValidateShapeIdReferenceInStopTime(mockDataRepo, mockResultRepo, mockLogger);
+
+        underTest.execute();
+
+        verify(mockLogger, times(1)).info(ArgumentMatchers.eq(
+                "Validating rule 'E034 - `shape_id` must be provided and valid when" +
+                        " stop_times.shape_dist_travelled is provided'" + System.lineSeparator()));
+
+        verify(mockDataRepo, times(1)).getStopTimeAll();
+        verify(mockDataRepo, times(1))
+                .getStopTimeByTripId(ArgumentMatchers.eq("trip id"));
+        verify(mockDataRepo, times(1)).getTripById("trip id");
+        verify(mockDataRepo, times(1)).getShapeById("non existing shape id");
```

Comment:
If it is valid, should the name be `shape id` instead of `non existing shape id`?

Reply:
good catch! Modified in f6c533bc1e8f5a1c1bc035603124f46165623c4f

Reply:
good catch! Modified in f6c533bc1e8f5a1c1bc035603124f46165623c4f